src/Cardano/Wallet/Mnemonic.hs

```diff
@@ -0,0 +1,332 @@
+sentenceToRawString =
+    Basement.toBytes Basement.UTF8 . mnemonicSentenceToString Dictionary.english
+
+
+
+toUtf8String
+    :: Text
+    -> Basement.String
+toUtf8String =
+    Basement.fromString . unpack
+
+
+fromUtf8String
+    :: Basement.String
+    -> Text
+fromUtf8String =
+    pack . Basement.toList
+
+
+-- | The initial seed has to be vector or length multiple of 4 bytes and shorter
+-- than 64 bytes. Not that this is good for testing or examples, but probably
+-- not for generating truly random Mnemonic words.
+--
+-- See 'Crypto.Random.Entropy (getEntropy)'
+instance
+    ( ValidEntropySize n
+    , ValidChecksumSize n csz
+    ) => Arbitrary (Entropy n) where
+    arbitrary =
+        let
+            size = fromIntegral $ ambiguousNatVal @n
+            entropy =
+                mkEntropy  @n . B8.pack <$> vectorOf (size `quot` 8) arbitrary
+        in
+            either (error . show . UnexpectedEntropyError) id <$> entropy
+
+
+-- | Same remark from 'Arbitrary Entropy' applies here.
+instance
+    ( n ~ EntropySize mw
+    , mw ~ MnemonicWords n
+    , ValidChecksumSize n csz
+    , ValidEntropySize n
+    , ValidMnemonicSentence mw
+    , Arbitrary (Entropy n)
+    ) => Arbitrary (Mnemonic mw) where
+    arbitrary =
+        entropyToMnemonic <$> arbitrary @(Entropy n)
+
+
+instance (KnownNat csz) => Basement.Exception (MnemonicException csz)
+
+
+-- | To use everytime we need to show an example of a Mnemonic. This particular
+-- mnemonic is rejected to prevent users from using it on a real wallet.
+instance Default (Mnemonic 12) where
+    def =
+        let
+            wordsm =
+                [ "squirrel"
+                , "material"
+                , "silly"
+                , "twice"
+                , "direct"
+                , "slush"
+                , "pistol"
+                , "razor"
+                , "become"
+                , "junk"
+                , "kingdom"
+                , "flee"
+                ]
+
+            phrase = either (error . show) id
+                (mnemonicPhrase @12 (toUtf8String <$> wordsm))
+
+            sentence = either (error . show) id
+                (mnemonicPhraseToMnemonicSentence Dictionary.english phrase)
+
+            entropy = either (error . show) id
+                (wordsToEntropy @(EntropySize 12) sentence)
+        in Mnemonic
+            { mnemonicToSentence = sentence
+            , mnemonicToEntropy  = entropy
+            }
+
+
+instance
+    ( n ~ EntropySize mw
+    , mw ~ MnemonicWords n
+    , ValidChecksumSize n csz
+    , ValidEntropySize n
+    , ValidMnemonicSentence mw
+    , Arbitrary (Entropy n)
+    ) => FromJSON (Mnemonic mw) where
+    parseJSON =
+        parseJSON >=> (eitherToParser . mkMnemonic)
+
+
+instance ToJSON (Mnemonic mw) where
+    toJSON =
+        toJSON
+        . map (fromUtf8String . dictionaryIndexToWord Dictionary.english)
+        . unListN
+        . mnemonicSentenceToListN
+        . mnemonicToSentence
```

Comment:
I'd suggest not to bring any JSON dependencies at this stage and provide instead, serialization combinators that may be use to define JSON instances where it's relevant.

```hs
mnemonicToText :: Mnemonic mw -> [Text]
mnemonicToText = ...

-- or

unMnemonic :: Mnemonic mw -> [Text]
unMnemonic = ... 
```

Reply:
added `mnemonicToText` because used in executable. But left aeson as they are used in the tests

Reply:
Nope, the Aeson tests should also be removed :+1: 
They will probably be part of the API tests, but that's the responsibility of the API layer, not this module.

Reply:
aeson tests and ToJSON, FromJSON instances removed